agent/consul/state/acl_schema.go

```diff
@@ -276,15 +276,30 @@ func bindingRulesTableSchema() *memdb.TableSchema {
 				Name:         indexAuthMethod,
 				AllowMissing: false,
 				Unique:       false,
-				Indexer: &memdb.StringFieldIndex{
-					Field:     "AuthMethod",
-					Lowercase: true,
+				Indexer: indexerSingle{
+					readIndex:  indexFromQuery,
+					writeIndex: indexAuthMethodFromACLBindingRule,
```

Comment:
All the reads for this index need to use a Query object instead of passing a string value.